silkworm/silkrpc/json/types.cpp

```diff
@@ -31,22 +32,28 @@ namespace silkworm::rpc {
 
 using evmc::literals::operator""_address;
 
-void to_hex(char* hex_bytes, silkworm::ByteView bytes) {
+void to_hex(std::span<char> hex_bytes, silkworm::ByteView bytes) {
     static const char* kHexDigits{"0123456789abcdef"};
-    char* dest{&hex_bytes[0]};
+    if (bytes.size() * 2 + 2 > hex_bytes.size()) {
+        throw std::invalid_argument("to_hex: hex_bytes too small");
+    }
+    char* dest = hex_bytes.data();
     *dest++ = '0';
     *dest++ = 'x';
     for (const auto& b : bytes) {
         *dest++ = kHexDigits[b >> 4];    // Hi
         *dest++ = kHexDigits[b & 0x0f];  // Lo
     }
+    *dest = '\0';
 }
 
-std::size_t to_hex_no_leading_zeros(char* hex_bytes, silkworm::ByteView bytes) {
+std::size_t to_hex_no_leading_zeros(std::span<char> hex_bytes, silkworm::ByteView bytes) {
     static const char* kHexDigits{"0123456789abcdef"};
-
-    char* dest{&hex_bytes[0]};
     size_t len = bytes.length();
+    if (len * 2 + 2 > hex_bytes.size()) {
```

Comment:
This should be `len * 2 + 2 + 1` to count also for the terminator character '\0'

Reply:
yes